### DISCUSSION TITLE
How can we generate Random Salt of 32 bytes in Rhino JS

### DISCUSSION THREAD

**Discussion by Karan-nayyar** [mozilla/rhino#1174] [Q&A] [answered]
I am trying to generate a random salt of 32 bytes size. But my JS engine Rhino 1.7.13 doesn't support SecureRandom class.

Below is the code snippet of the same.

`function getSalt() {
    var random = new SecureRandom();
    var salt1 = new Array(32);
    random.nextBytes(salt1);
    return salt1;`

Error logged as below.

`java.util.concurrent.ExecutionException: javax.script.ScriptException: ReferenceError: "SecureRandom" is not defined.
`
Also, rhino js engine does not allow any import or load of external library. Is there a way we can generate secure random salts in Rhino?

**Comment by tuchida** [comment, marked as answer]
This worked in my environment.
```js
var random = new java.security.SecureRandom();
var salt1 = java.lang.reflect.Array.newInstance(java.lang.Byte.TYPE, 32);
random.nextBytes(salt1);
salt1.join(); // -69,-43,62,15,-48,-78,-68,-10,-74,-108,-43,44,-43,-108,27,8,41,102,91,-64,-88,-81,107,-46,-99,24,-64,85,11,89,52,21
```
